## Goal: TMEM129 related 
- TCGA BLCA T/N rs2236786 GT in T-drive:
- Get junction readcount TMEM129 from TCGA data from Recount3 package

## Code: 

**Extract GT from TCGA BLCA vcf generated by Oscar** 

```R

library(vcfR)
library(tidyr)
library(reshape2)
library(stringr) 
library(textclean)
library(xlsx)
library(dplyr)

vcf.file.tmp <- read.vcfR("/Volumes/ifs/DCEG/Branches/LTG/Prokunina/TCGA_data/TCGA_WGS_BLCA_hg38/projects/mapBC/multi_sample_output_Tumor.vcf")
#vcf.file.tmp <- read.vcfR("/Volumes/ifs/DCEG/Branches/LTG/Prokunina/TCGA_data/TCGA_WGS_BLCA_hg38/projects/mapBC/multi_sample_output_Blood.vcf")

vcf.chr4 <- vcf.file.tmp[getCHROM(vcf.file.tmp) == "chr4", ]

# get and format genotypes, set "return.alleles = TRUE" can switch 0|1 to A|G
vcf_genotypes.tmp <- extract.gt(
  vcf.chr4,
  element = "GT",
  mask = FALSE,
  as.numeric = FALSE,
  return.alleles = TRUE,
  IDtoRowNames = TRUE,
  extract = TRUE,
  convertNA = TRUE
)

geno_df.tmp <- as.data.frame(t(vcf_genotypes.tmp))
geno_df.tmp$ID_person <- row.names(geno_df.tmp)
df_sub <- geno_df.tmp %>% select(ID_person,chr4_1717567)

df_sub$full_ID <-df_sub$ID_person 
df_sub$ID_person <- sub("^(([^-]+-){2}[^-]+).*", "\\1", df_sub$full_ID)
df_sub <- df_sub[,c(1,3,2)]
names(df_sub)[3] <- "rs2236786_chr4_1717567"

# write.csv(df_sub,'/Volumes/ifs/DCEG/Branches/LTG/Prokunina/TCGA_data/Data_for_mapBC_project/mapBC_genotypes_rs2236786_TCGA_BLCA_Tumor.csv',row.names = F)


```


