## Goal: TMEM129 related 
- TCGA BLCA T/N rs2236786 GT in T-drive:
- Get junction readcount TMEM129 from TCGA data from Recount3 package

## Code: 

**Extract GT from TCGA BLCA vcf generated by Oscar** 

```R

library(vcfR)
library(tidyr)
library(reshape2)
library(stringr) 
library(textclean)
library(xlsx)
library(dplyr)

vcf.file.tmp <- read.vcfR("/Volumes/ifs/DCEG/Branches/LTG/Prokunina/TCGA_data/TCGA_WGS_BLCA_hg38/projects/mapBC/multi_sample_output_Tumor.vcf")
#vcf.file.tmp <- read.vcfR("/Volumes/ifs/DCEG/Branches/LTG/Prokunina/TCGA_data/TCGA_WGS_BLCA_hg38/projects/mapBC/multi_sample_output_Blood.vcf")

vcf.chr4 <- vcf.file.tmp[getCHROM(vcf.file.tmp) == "chr4", ]

# get and format genotypes, set "return.alleles = TRUE" can switch 0|1 to A|G
vcf_genotypes.tmp <- extract.gt(
  vcf.chr4,
  element = "GT",
  mask = FALSE,
  as.numeric = FALSE,
  return.alleles = TRUE,
  IDtoRowNames = TRUE,
  extract = TRUE,
  convertNA = TRUE
)

geno_df.tmp <- as.data.frame(t(vcf_genotypes.tmp))
geno_df.tmp$ID_person <- row.names(geno_df.tmp)
df_sub <- geno_df.tmp %>% select(ID_person,chr4_1717567)

df_sub$full_ID <-df_sub$ID_person 
df_sub$ID_person <- sub("^(([^-]+-){2}[^-]+).*", "\\1", df_sub$full_ID)
df_sub <- df_sub[,c(1,3,2)]
names(df_sub)[3] <- "rs2236786_chr4_1717567"

# write.csv(df_sub,'/Volumes/ifs/DCEG/Branches/LTG/Prokunina/TCGA_data/Data_for_mapBC_project/mapBC_genotypes_rs2236786_TCGA_BLCA_Tumor.csv',row.names = F)


```


**Extract junction count of TMEM129 from Recount3 package**

- Need to run in Biowulf

```R


library(recount3)
library(snapcount)
library(megadepth)
library(dplyr)
library(tidyr)

# the temp file is located in leec20/.cache/R/recount3
# needd to clean if too many?
# Note the TCGA file need to used in BIouwlf Rstudio, it's too large

human_projects <- available_projects()

# get the TCGA and GTEx 

#gtex <- subset(human_projects, file_source == "gtex" & project_type == "data_sources")
TCGA <- subset(human_projects, file_source == "tcga" & project_type == "data_sources")

list_TCGA <- TCGA$project


# for ( i in list_TCGA){
  
  df <- data.frame()
  
  i <- "BLCA"
  target_tissue  <- TCGA %>% filter(project == i)
  
  rse_jxn_b <- create_rse(
    target_tissue,
    type = "jxn"
  )
  
  # To find the TCGA ID, it's in the package 
  TCGAIDmatch <- rse_jxn_b@colData@listData %>% as.data.frame()
  TCGAIDmatch <- TCGAIDmatch[,c(2,4)]
  
  matrix_data <- as.matrix(assay(rse_jxn_b))
  
  df <- as.data.frame(matrix_data)
  colnames(df) <- TCGAIDmatch$tcga.tcga_barcode[match(colnames(df), TCGAIDmatch$external_id)]
  
  
  
  # TMEM129 region in hg38
  # chr4:1715712-1721408
  # chr4:1715936-1721342
  
  df$POS <- rownames(df)
  
  df_chr4 <- df[which(grepl("chr4:",df$POS)),]
  df_chr4 <- df_chr4 %>%  separate(POS, into = c("chr", "start", "end",'strand'), sep = "[:-]")
  df_chr4$strand <- gsub('chr4:[0-9]+-[0-9]+:','',rownames(df_chr4))
  
  df_chr4$start <- as.numeric(df_chr4$start)
  df_chr4$end <- as.numeric(df_chr4$end)
  
  df_chr4 <- df_chr4 %>% 
    filter(start >= 1715952 & start <= 1721358 & 
             end >= 1715952 & end <= 1721358)
  
  
  df_chr4 <- df_chr4[,c(434:437,1:433)]
  
  # remove sum = 1
  
  df_chr4 <- df_chr4 %>% 
    mutate(SUM = rowSums(select(., 5:437)))
  
  df_chr4 <- df_chr4 %>% filter(SUM > 1)
  df_chr4 <- df_chr4 %>% filter(df_chr4$start > 1717426)
  # relocate column
  df_chr4 <- df_chr4 %>% relocate(SUM,.before = 1)
  
  df_chr4$location <- paste0('chr4:',df_chr4$start,"-",df_chr4$end)
  df_chr4 <- df_chr4[,c(grep("location",names(df_chr4),value = T),grep("TCGA",names(df_chr4),value = T))]
  
  
  df_chr4.t <- as.data.frame(t(df_chr4))
  names(df_chr4.t) <- df_chr4.t[1,]
  df_chr4.t <- df_chr4.t[-1,]
  df_chr4.t[] <- lapply(df_chr4.t[], as.numeric)
  df_chr4.t <- df_chr4.t %>% mutate(TCGA_ID=rownames(df_chr4.t),.before = 1)
  
  write.table(df_chr4.t,paste0("~/Desktop/chr15_CHRNA5_bladder_project/Recount3_for_GTEx_and_TCGA/TCGA_JC/",i,"_C5iso_jc.csv"),col.names = T,row.names = F,sep = ',',quote = F)
  
# }


## To do list

- finding TPM of TCGA/GTEx of TMEM129 isoform, can we use UCSC Xena data we dowlaoded for TERT paper?
  The locatation of file:
`/Volumes/ifs/DCEG/Branches/LTG/Prokunina/TCGA_data/Original source data Expression/UCSC Toil RNA-seq Recompute`

- IMVIGOR210 dataset
We have raw count, patient info.
What data should we extract next?








```

